### PR TITLE
fix!: remove configstore

### PR DESCRIPTION
**Pull request by shaffeeullah**
Fixes: https://github.com/googleapis/gcs-resumable-upload/issues/347

### CODE REVIEW

File: src/gcs-resumable-upload.ts [hunk 1]
```diff
@@ -887,30 +820,6 @@ export class Upload extends Pumpify {
...
-      // this resumable upload is unrecoverable (bad data or service error).
-      //  -
-      //  https://github.com/googleapis/gcs-resumable-upload/issues/15
-      //  -
-      //  https://github.com/googleapis/gcs-resumable-upload/pull/16#discussion_r80363774
-      if (resp && resp.status === TERMINATED_UPLOAD_STATUS_CODE) {
-        this.restart();
-        return;
-      }
-
```
Comment (danielbankhead) [inline src/gcs-resumable-upload.ts:913]: Why is this block getting removed?

**Review by danielbankhead — COMMENTED** [review]
Such a therapeutic PR

File: src/gcs-resumable-upload.ts (hunk 1 above)
Comment (shaffeeullah) [reply src/gcs-resumable-upload.ts:913]: We don't want to retry 404s. This was the case of a stale URI. `TERMINATED_UPLOAD_STATUS_CODE` is not a retryable error code.

**Comment by danielbankhead** [comment]
I think `xdg-basedir` can be removed in this PR too

**Review by danielbankhead — APPROVED** [review]

**Review by ddelgrosso1 — APPROVED** [review]

**Comment by rhodgkins** [comment]
Thanks @ddelgrosso1, out of interest how do resumable uploads work now? Are they only resumable for the lifetime of the nodejs process?

**Comment by ddelgrosso1** [comment]
@rhodgkins apologies for the delay in responding. Yes that is correct they are only resumable during the lifetime of a process now. We realized during our investigation into whether or not configStore was needed that there existed the possibility of data corruption in the old methodology. ConfigStore was only being used to store the first 16 bytes of an uploaded file which is a poor way to determine if the files are the same. Coupled with the numerous issues with having to cache on disk we made the decision to remove configStore entirely. Now, during the lifetime of a process a resumable upload will resume from the spot of interruption.

**Pull request merged**